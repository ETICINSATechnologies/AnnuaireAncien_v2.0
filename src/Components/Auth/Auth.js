class Auth {
    static connect(jwt){
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(){
        sessionStorage.removeItem('token')
    }

    static isConnected(){
        return !!sessionStorage.getItem('token');
    }

    static getToken() {
        return sessionStorage.getItem('token')
    }

    static getPayload(){
        return sessionStorage.getItem('token');
    }

    static addCorrectButton(buttons){
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
