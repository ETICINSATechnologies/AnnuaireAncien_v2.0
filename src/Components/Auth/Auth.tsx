class Auth {
    static connect(jwt: string) {
        sessionStorage.setItem('token', jwt);
    }

    static disconnect(): void {
        sessionStorage.removeItem('token')
    }

    static isConnected(): boolean {
        return !!sessionStorage.getItem('token');
    }

    static getToken(): string {
        return sessionStorage.getItem('token') as string;
    }

    static addCorrectButton(buttons: Array<string>) {
        if (this.isConnected())
            buttons.push("disconnection");
        else
            buttons.push("connection");
        return buttons;
    }
}

export default Auth;
