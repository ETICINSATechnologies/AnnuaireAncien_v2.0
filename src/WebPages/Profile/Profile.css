.Profile {
    display: grid;

    width: 100%;
    height: 100%;
    grid-row: 3;
    grid-column: 1;
    justify-self: center;
    vertical-align: middle;
    margin: auto;
    grid-template-columns: 2fr 1fr;
}
