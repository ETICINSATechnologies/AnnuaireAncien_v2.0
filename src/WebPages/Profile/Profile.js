import React, {Component} from 'react';
import './Profile.css';
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import Redirect from "react-router-dom/es/Redirect";
import Auth from "../../Components/Auth/Auth";
import ProfileForm from "../../Components/ProfileForm/ProfileForm";
import PositionForm from "../../Components/PositionForm/PositionForm";

class Profile extends Component {
    constructor(props) {
        super(props);
        this.state = {
            status: 'pending', // 'connected' 'not_authenticate'
            modifyEnabled: false,
            positions:{},
            currentPositions:[],
            currentPositionsBackup:[],
            info: {
                firstName: '',
                lastName: '',
                email: '',
                telephone: '',
                department: {
                    id:'',
                    label:'',
                    name:'',
                },
                company: ''
            },
        }
    }

    componentDidMount() {
        if (!Auth.isConnected())
            this.setState({status: 'not_authenticate'});
        else
        {
            this.setState({status: "connected"});
            fetch('api/v1/core/member/me', {
                headers: {
                    Authorization: Auth.getToken()
                }
            })
                .then(res => res.json())
                .then((result) => {
                    this.setState({
                        info: result,
                        currentPositions : result.positions,
                        currentPositionsBackup : JSON.parse(JSON.stringify(result.positions))
                    });
                })
        }
    }


    updateInfo(data){
        data.positions=this.state.currentPositions;
        let i;
        for (i=0;i<data.positions.length;++i){
            data.positions[i].year=parseInt(data.positions[i].year,10)
        }
        fetch('api/v1/core/member/'+this.state.info.id, {
            method: 'PUT',
            headers: {
                'Authorization': Auth.getToken(),
                'Content-Type': 'application/json'
            },

            body: JSON.stringify(data)
        })
            .then(res => {
                if (res.status === 200) {
                        res.json()
                        .then((result) => {
                            this.setState({
                                info: result,
                                currentPositions : result.positions,
                                currentPositionsBackup : JSON.parse(JSON.stringify(result.positions))
                            });
                        })
                } else {
                    this.resetFields();
                    console.log('update fail');
                }
            });
        //disable modifications
        if (this.state.modifyEnabled) this.setModify(false);
    }

    setModify(state){
        this.setState({
            modifyEnabled: state,
        });
    }

    updatePositions(positions){
        this.setState({
            currentPositions : positions,
        });
    }

    resetFields(){
        this.setState({
            info : this.state.info,
            modifyEnabled : false,
            currentPositions : JSON.parse(JSON.stringify(this.state.currentPositionsBackup))
        });
    }

    render() {
        let activeButton = ["home"];
        if (this.state.status === 'not_authenticate')
            return <Redirect to='/'/>;

        activeButton.push('search');
        activeButton = Auth.addCorrectButton(activeButton);

        return (
            <React.Fragment>
                <Header/>
                <Nav buttons={activeButton}> </Nav>
                <section className="Profile">
                    <ProfileForm info={this.state.info} update={this.updateInfo.bind(this)}
                                 modifyEnabled={this.state.modifyEnabled}
                                 setModify={this.setModify.bind(this)} resetFields={this.resetFields.bind(this)}
                    />
                    <PositionForm positions={this.state.positions} modifyEnabled={this.state.modifyEnabled}
                                  currentPositions={this.state.currentPositions}
                                  updatePositions={this.updatePositions.bind(this)}
                    />
                </section>
            </React.Fragment>
        );
    }
}

export default Profile;
