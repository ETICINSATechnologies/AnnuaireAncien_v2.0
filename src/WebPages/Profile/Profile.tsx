import React, {Component} from 'react';
import {Redirect} from 'react-router-dom';

import './Profile.css';
import Header from "../../Components/Header/Header";
import Nav from "../../Components/Nav/Nav";
import Auth from "../../Components/Auth/Auth";
import ProfileForm from "../../Components/ProfileForm/ProfileForm";
import PositionForm from "../../Components/PositionForm/PositionForm";

import {Member, MemberInterface, defaultMember} from "../../Model/Member";
import {MemberPosition} from "../../Model/MemberPosition";


interface ProfileState {
    status: string
    modifyEnabled: boolean
    update: boolean
    updateSucceed: boolean
    previousMember: Member
    member: Member
}


class Profile extends Component<{}, ProfileState> {
    state = {
        status: 'pending', // 'connected' 'not_authenticate'
        modifyEnabled: false,
        update: false,
        updateSucceed: false,
        previousMember: defaultMember,
        member: defaultMember
    };

    componentDidMount() {
        if (!Auth.isConnected())
            this.setState({status: 'not_authenticate'});
        else {
            this.setState({status: "connected"});
            fetch('member/me', {
                headers: {
                    Authorization: Auth.getToken()
                }
            })
                .then(res => res.json())
                .then((result) => {
                    let member = new Member(JSON.parse(JSON.stringify(result)));
                    let prevMember = new Member(JSON.parse(JSON.stringify(result)));
                    this.setState({
                        previousMember: prevMember,
                        member: member
                    });
                })
        }
    }

    updateMember = () => {
        fetch('member/' + this.state.member.id, {
            method: 'PUT',
            headers: {
                'Authorization': Auth.getToken(),
                'Content-Type': 'application/json'
            },

            body: JSON.stringify(this.state.member.update())
        })
            .then(res => {
                if (res.status === 200) {
                    res.json()
                        .then((result) => {
                            let member = new Member(JSON.parse(JSON.stringify(result)));
                            let prevMember = new Member(JSON.parse(JSON.stringify(result)));
                            this.setState({
                                previousMember: prevMember,
                                member: member,
                                update: true,
                                updateSucceed: true,
                                modifyEnabled: false
                            });
                        })
                } else {
                    this.setState({
                        update: true,
                        updateSucceed: false
                    });
                }
            });
    };

    updateMemberPassword = (pass : string) => {

        let member= this.state.previousMember;
        member.password=pass;

        fetch('member/' + this.state.member.id, {
            method: 'PUT',
            headers: {
                'Authorization': Auth.getToken(),
                'Content-Type': 'application/json'
            },

            body: JSON.stringify(member.update())
        })
            .then(res => {
                if (res.status === 200) {
                    res.json()
                        .then(() => {
                            this.setState({
                                update: true,
                                updateSucceed: true,
                            });
                        })
                } else {
                    this.setState({
                        update: true,
                    });
                }
            });
    };

    enableModification = () => {
        let member = new Member(JSON.parse(JSON.stringify(this.state.previousMember)));
        this.setState({
            update: false,
            modifyEnabled: !this.state.modifyEnabled,
            member: member
        })
    };

    resetFields = () => {
        let member = new Member(JSON.parse(JSON.stringify(this.state.previousMember)));
        this.setState({
            member : member,
            modifyEnabled : false,
        });
    };

    updateMemberPositions = (mPositions: MemberPosition[]) => {
        this.setState({
            member: new Member({
                ...this.state.member,
                positions: mPositions
            } as MemberInterface)
        });
    };

    modifyMember = (member: Member) => {
        this.setState({
            member: member
        })
    };

    render() {
        let activeButton = ["home"];
        if (this.state.status === 'not_authenticate')
            return <Redirect to='/'/>;

        activeButton.push('search');
        activeButton = Auth.addCorrectButton(activeButton);

        return (
            <React.Fragment>
                <Header/>
                <Nav buttons={activeButton}/>
                <section className="Profile">
                    <ProfileForm member={this.state.member} modifyEnabled={this.state.modifyEnabled}
                                 update={this.state.update} updateSucceed={this.state.updateSucceed}
                                 modifyMember={this.modifyMember} updateMember={this.updateMember}
                                 updateMemberPassword={this.updateMemberPassword}
                                 enableModification={this.enableModification} resetFields={this.resetFields}/>
                    <PositionForm memberPositions={this.state.member.positions} modifyEnabled={this.state.modifyEnabled}
                                  updateMemberPositions={this.updateMemberPositions}/>
                </section>
            </React.Fragment>
        );
    }
}

export default Profile;
