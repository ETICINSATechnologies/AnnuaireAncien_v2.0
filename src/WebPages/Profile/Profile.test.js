import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from "react-router-dom";

import Profile from './Profile';
import Auth from '../../Components/Auth/Auth';

Enzyme.configure({adapter: new Adapter()});

describe( 'Profile component', () => {
    it('renders without crashing', () => {
            const componentDiv = document.createElement('div');
            ReactDOM.render(shallow(
                <Router>
                    <Profile/>
                </Router>
            )
            , componentDiv);
            ReactDOM.unmountComponentAtNode(componentDiv);
        }
    );

    it('Profile unit test', () => {
        Auth.isConnected = jest.fn(() => true);
        expect(
            shallow(<Profile/>)
            .find('section.Profile')
            .length
        ).toBe(1)

    });

    it('Profile not connected snapshot test', () => {
        Auth.isConnected = jest.fn(() => false);
        const component = renderer.create(
            <Router>
                <Profile/>
            </Router>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Profile connected snapshot test', () => {
        Auth.isConnected = jest.fn(() => true);
        const component = renderer.create(
            <Router>
                <Profile/>
            </Router>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

});