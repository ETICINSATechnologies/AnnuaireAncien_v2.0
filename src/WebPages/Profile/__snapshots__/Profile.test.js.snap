// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component Profile connected snapshot test 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens 
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/"
        onClick={[Function]}
      >
        accueil
      </a>
      <a
        className="button"
        href="/search"
        onClick={[Function]}
      >
        rechercher
      </a>
      <a
        className="button"
        href="/"
        onClick={[Function]}
      >
        se déconnecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
      onClick={[Function]}
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <section
    className="Profile"
  >
    <form
      className="ProfileForm"
    >
      <div
        className="header_container"
      >
        <h1>
          Appuyer sur le crayon pour modifier
        </h1>
        <img
          alt="Profile"
          className="profilePicture"
          src="default_woman.svg"
        />
        <img
          alt="Modifier/Annuler"
          className="deleteCancelPicture"
          onClick={[Function]}
          src="edit.png"
        />
      </div>
      <div
        className="field_container"
      >
        <p>
           Téléphone 
        </p>
        <p
          className="right needed"
        >
           Nom 
        </p>
        <input
          className="telephone"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="lastName"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
        <p>
           Département 
        </p>
        <p
          className="right p_info needed"
        >
           Prénom 
        </p>
        <div
          className="DropDown"
        >
          <select
            className="department"
            disabled={true}
            onChange={[Function]}
            value={0}
          >
            <option
              value={0}
            >
               Choisir une option
            </option>
          </select>
        </div>
        <input
          className="firstName"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
        <p
          className="needed"
        >
           Adresse mail 
        </p>
        <p>
           Travaille chez 
        </p>
        <input
          className="email"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="company"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="button_container"
      />
      <div
        className="modal"
      />
    </form>
    <div
      className="position_form"
    >
      <h1>
        Postes occupés à ETIC
      </h1>
      <div
        className="positions"
      >
         
         
      </div>
      <input
        className="input_button add_position hidden"
        onClick={[Function]}
        type="button"
        value="Ajouter"
      />
    </div>
  </section>,
]
`;

exports[`Profile component Profile not connected snapshot test 1`] = `
Array [
  <header
    className="Header"
  >
    <a
      className="header_a"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="home_logo"
        src="home_logo.png"
      />
      <h1
        className="title"
      >
         Annuaire des Anciens 
      </h1>
    </a>
  </header>,
  <nav
    className="Nav"
  >
    <div
      className="Menu"
    >
      <a
        className="button"
        href="/"
        onClick={[Function]}
      >
        accueil
      </a>
      <a
        className="button"
        href="/search"
        onClick={[Function]}
      >
        rechercher
      </a>
      <a
        className="button"
        href="/connection"
        onClick={[Function]}
      >
        se connecter
      </a>
    </div>
    <a
      className="help_link"
      href="/help"
      onClick={[Function]}
    >
      <img
        alt=""
        className="help_icon"
        src="help_icon.png"
      />
    </a>
  </nav>,
  <section
    className="Profile"
  >
    <form
      className="ProfileForm"
    >
      <div
        className="header_container"
      >
        <h1>
          Appuyer sur le crayon pour modifier
        </h1>
        <img
          alt="Profile"
          className="profilePicture"
          src="default_woman.svg"
        />
        <img
          alt="Modifier/Annuler"
          className="deleteCancelPicture"
          onClick={[Function]}
          src="edit.png"
        />
      </div>
      <div
        className="field_container"
      >
        <p>
           Téléphone 
        </p>
        <p
          className="right needed"
        >
           Nom 
        </p>
        <input
          className="telephone"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="lastName"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
        <p>
           Département 
        </p>
        <p
          className="right p_info needed"
        >
           Prénom 
        </p>
        <div
          className="DropDown"
        >
          <select
            className="department"
            disabled={true}
            onChange={[Function]}
            value={0}
          >
            <option
              value={0}
            >
               Choisir une option
            </option>
          </select>
        </div>
        <input
          className="firstName"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
        <p
          className="needed"
        >
           Adresse mail 
        </p>
        <p>
           Travaille chez 
        </p>
        <input
          className="email"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="company"
          disabled={true}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="button_container"
      />
      <div
        className="modal"
      />
    </form>
    <div
      className="position_form"
    >
      <h1>
        Postes occupés à ETIC
      </h1>
      <div
        className="positions"
      >
         
         
      </div>
      <input
        className="input_button add_position hidden"
        onClick={[Function]}
        type="button"
        value="Ajouter"
      />
    </div>
  </section>,
]
`;
